File: Python/1557.py
n = int(input())
mat = []
cont = 1
aux = 1
for i in range(0,n+1):
    linha = []
    for j in range(0,n+1):
        linha.append(cont)
        cont *= 2
    cont = aux*2
    aux *= 2
    mat.append(linha)

for i in range(0,n+1):
    for j in range (0,n+1):
        print(mat[i][j],end = "\t")
    print("\r")
    

File: C/1557.c
#include<stdio.h>
int main()
{
    int x,i,j;
    while(1)
    {
        scanf("%d",&x);
        if(x==0)
        {
            return 0;
        }
        int mat[x][x],cont=1,aux=1;
        for(i=0;i<x;i++)
        {
            for(j=0;j<x;j++)
            {
                mat[i][j]=cont;
                cont*=2;
            }
            cont=aux*2;
            aux*=2;
        }
        for(i=0;i<x;i++)
        {
            for(j=0;j<x;j++)
            {
                printf("%d",mat[i][j]);
                if(j > 0)
	            {
	            	printf("\t");
	            }
            }
            printf("\n");
        }
        printf("\n");
    }
    return 0;
}
